exercicios/projeto-casa/index.js

```diff
@@ -0,0 +1,147 @@
+          
+        })
+        listaClientesbanco.map((conta, index) =>{
+            if (conta.cpf_cliente != cpfCliente){
+                listaClientesbanco[index].endereco = novoendereco
+            }
+            return res.status(200).json({mensagem: 
+                `O enderenco do cliente é: ${ClienteComIdValido.nome_cliente} Foi atualizado para   ${ClienteComIdValido.atualizarend} `})
+        })
+    }
+    return res.status(404).json({mensagem: 'cliente não encontrado'});
+})
++
+//- Fazer depósitos / pagamentos usando o saldo de sua conta (FEITO, Funcinou tentei fazer com req.params mas não rolou, não se poderia fazer com req.query para)
+app.patch('/deposito', (req, res ) => { 
+    const SolicitaCpfcliente = req.query.cpf_cliente // usei o query para consultar no banco se o cpf era válido
+    const {valorDeposito} = req.body
+    const ExisteCliente = listaClientesbanco.find((cliente) => cliente.cpf_cliente == SolicitaCpfcliente)
+    if (ExisteCliente){
+       const NovoSaldoConta ={ 
+        ...ExisteCliente.conta,
+        saldo: ExisteCliente.conta.saldo + valorDeposito,
+    } 
+    listaClientesbanco.map((cliente, index)=>{
+      if (cliente.cpf_cliente == SolicitaCpfcliente){
+        return (listaClientesbanco[index] = NovoSaldoConta)
+      }
+    })
+    res.status(202).json(NovoSaldoConta)
+} 
+
+    return res.status(404).json({ messagem: 'Usuário não existe' }); 
+})
+// rota para saque(pagamento)(FEITO)
+app.patch('/saque', (req, res) =>{
+    const VerificaCpf = req.query.cpf_cliente
+    const {Valordosaque} = req.body
+    const Existecliente = listaClientesbanco.find((cliente) => cliente.cpf_cliente == VerificaCpf)
+    if (Existecliente.conta.saldo >= Valordosaque){
+        const Saque = {
+            ...Existecliente.conta,
+            Saldo: Existecliente.conta.saldo - Valordosaque,
+        }
+        listaClientesbanco.map((cliente, index) =>{
+            if (cliente.cpf_cliente == VerificaCpf)
+            return (listaClientesbanco[index] = Valordosaque)
+        } )
+        res.status(202).json(Saque)
+    }
+return res.status(404).json({mensagem: "Seu saldo é insuficiente para o saque!"})
+})
+//- Encerrar contas de clientes (encerei a conta por cpf )(feito)
+app.delete('/clientes/:cpf_cliente', (req, res) =>{
+    const cpfCliente = req.params.cpf_cliente
+    const cpfvalido = listaClientesbanco.find(usario => usario.cpf_cliente == cpfCliente)
+    if (cpfvalido){
+    listaClientesbanco.map((usario, index) =>{
+        if (usario.cpf_cliente == cpfCliente)
+        return listaClientesbanco.slice(index,1)
+    })
+    return res.status(202).json({mensagem:"cliente apagado com sucesso", usario: cpfvalido})
+}
+return res.status(404).json({mensagem: 'Este cpf não é válido, tente novamente '})
+})
+//- Conseguir Filtrar os clientes do banco pelo seu nome,por saldo...(DONE) alguns tipos de filtros
+app.get('/clientes', (req, res) =>{ 
+    const filtroNome = req.query.nome_cliente
```

Comment:
O `destruct` também funciona para o query
Ex.:
```javascript
const {nome_cliente, cpf , data_nascimento} = req.query
```

Reply:
eu usei o destruct depois, esse foi o primeiro que fiz

Reply:
Pode utilizar quantas vezes for necessário.